### DISCUSSION TITLE
-fmath-errno breaks compilation on RX 6700XT (Fedora 40)

### DISCUSSION THREAD

**Discussion by pahaze** [YellowRoseCx/koboldcpp-rocm#61] [General]
After one of the recent commits, Makefile is edited to have `-fmath-errno` in it as a CFLAG and CXXFLAG. It does not allow compilation giving undefined errors for `logf` and `expf`.

```
lld: error: undefined hidden symbol: logf
>>> referenced by /tmp/tsembd-gfx1030-34ece1.o:(timestep_embedding_f32(float const*, float*, int, int, int))
>>> referenced by /tmp/tsembd-gfx1030-34ece1.o:(timestep_embedding_f32(float const*, float*, int, int, int))

lld: error: undefined hidden symbol: expf
>>> referenced by /tmp/tsembd-gfx1030-34ece1.o:(timestep_embedding_f32(float const*, float*, int, int, int))
>>> referenced by /tmp/tsembd-gfx1030-34ece1.o:(timestep_embedding_f32(float const*, float*, int, int, int))
clang: error: amdgcn-link command failed with exit code 1 (use -v to see invocation)
make: *** [Makefile:274: ggml/src/ggml-cuda/tsembd.o] Error 1
```

Removing `-fmath-errno` allowed the build to work as normal.

**Comment by SpoiledCoding** [comment]
Thank you-- I ran into this same error and a couple hours of google didn't help. 

So I spent a couple more hours to set up ssh on my machine and committed the change to remove the flags. :)

[PR here! ](https://github.com/YellowRoseCx/koboldcpp-rocm/pull/62)

**Reply by SpoiledCoding** [reply]
Closed the PR, comment from YellowRoseCx [explained](https://github.com/YellowRoseCx/koboldcpp-rocm/pull/62#issuecomment-2283288106) another workaround and the root cause.

**Comment by YellowRoseCx** [comment]
Here's the official response from AMD:
> Hi @YellowRoseCx, I've spoken to the internal team about this. ROCm does not support the `-fmath-errno` flag on the GPU, so passing this flag to clang is causing your issues. Previously, we were pulling in math functions which did not implement errno to begin with, so this was failing silently. We're looking at finding a way to allow passing `-fmath-errno` to clang without breaking things so that builds like yours can function as they used to. As a workaround for now, you can pass `-Xarch_host -fmath-errno` to clang instead of `-fmath-errno`, which will enable this flag on the host only.


https://github.com/ROCm/ROCm/issues/3573#issuecomment-2307185904

**Reply by pahaze** [reply]
Nice! Glad to see an official response from AMD, as well as something more official to use than just simply removing the flag.
